### PR TITLE
Add debug logs in decryption and ckgo producer/consumer creation

### PR DESCRIPTION
**Pull request by MuweiHe**
Release Notes
-------------

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- PLACEHOLDER

Checklist
---------
- [x] Leave this box unchecked if features are not yet available in production

What
----

Add debug logs in decryption and ckgo producer/consumer creation. Windows users have been reporting see: Error: failed to create consumer: decryptbytes: procdecryptdata: The parameter is incorrect.
Adding logs in ckgo as well, as this doesn't really look like a dpapi error. Possibilities: sasl authentication?

References
----------

https://github.com/confluentinc/cli/issues/2412

Test & Review
-------------

### CODE REVIEW

File: pkg/secret/encryption_other.go [hunk 1]
```diff
@@ -99,6 +100,7 @@ func Decrypt(username, encrypted string, salt, nonce []byte) (string, error) {
 	if len(nonce) != NonceLength {
 		return "", fmt.Errorf(errors.IncorrectNonceLengthErrorMsg)
 	}
+	log.CliLogger.Debugf("Decrypting secret: %s", cipherText)
```
Comment (MuweiHe) [inline pkg/secret/encryption_other.go:103]: only added debugf for decrypting not encrypting as I don't want to print plaintext secrets in logs.

**Review by MuweiHe — COMMENTED** [review]
@brianstrauch @sgagniere plz let me you wdyt~~

File: internal/kafka/confluent_kafka.go [hunk 2]
```diff
@@ -127,6 +127,7 @@ func retrieveUnsecuredToken(e ckafka.OAuthBearerTokenRefresh, tokenValue string)
 func newProducer(kafka *config.KafkaClusterConfig, clientID, configPath string, configStrings []string) (*ckafka.Producer, error) {
 	configMap, err := getProducerConfigMap(kafka, clientID)
 	if err != nil {
+		log.CliLogger.Warn("Failed to get Confluent Kafka producer configuration map.")
 		return nil, err
```
Comment (brianstrauch) [inline internal/kafka/confluent_kafka.go:131]: Should we wrap these into the same error?

Comment (brianstrauch) [reply internal/kafka/confluent_kafka.go:131]: Something like `failed to get producer configuration: %w`

Comment (MuweiHe) [reply internal/kafka/confluent_kafka.go:131]: The error will later be returned and shown as "failed to create producer/consumer"

Comment (brianstrauch) [reply internal/kafka/confluent_kafka.go:131]: maybe just "failed to get configuration" in that case?

File: pkg/errors/error_message.go [hunk 3]
```diff
@@ -38,6 +38,7 @@ const (
 	// kafka topic commands
 	FailedToCreateProducerErrorMsg    = "failed to create producer: %v"
 	FailedToCreateConsumerErrorMsg    = "failed to create consumer: %v"
+	FailedToGetConfigurationErrorMsg  = "failed to get configuration map: %v"
```
Comment (brianstrauch) [inline pkg/errors/error_message.go:41]: Let's use `%w` to wrap errors. This lets us unwrap them later if we need to!
```suggestion
	FailedToGetConfigurationErrorMsg  = "failed to get configuration map: %w"
```

**Review by brianstrauch — DISMISSED** [review]

**Review by sgagniere — APPROVED** [review]

**Pull request merged**